src/main/java/graphql/execution/instrumentation/ChainedInstrumentation.java

```diff
@@ -117,13 +117,30 @@ public InstrumentationState createState() {
     }
 
     @Override
-    public CompletableFuture<ExecutionResult> instrumentExecutionResult(CompletableFuture<ExecutionResult> executionResultFuture, InstrumentationExecutionParameters parameters) {
-        for (Instrumentation instrumentation : instrumentations) {
-            InstrumentationState state = getState(instrumentation, parameters.getInstrumentationState());
-            executionResultFuture = instrumentation
-                    .instrumentExecutionResult(executionResultFuture, parameters.withNewState(state));
+    public CompletableFuture<ExecutionResult> instrumentExecutionResult(ExecutionResult executionResult, InstrumentationExecutionParameters parameters) {
+        CompletableFuture<ExecutionResult> result = new CompletableFuture<>();
+
+        chainNthResult(0, executionResult, parameters, result);
+        return result;
+    }
+
```

Comment:
This pattern is interesting and a little hard to grok at first.  But if you want A -> B ->C I guess you have to